packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -315,6 +377,10 @@ export default util.createRule({
           objectText = getText(node.object);
           break;
 
+        case AST_NODE_TYPES.TSNonNullExpression:
+          // Not supported mixing with TSNonNullExpression
+          return '';
```

Comment:
Codecov is right, this isn't covered by any tests. Could you add one in please?

Reply:
I don't think it's reachable, I handled the `TSNonNullExpression` case twice just in case in 2 different functions just to be safe 😓

Reply:
I think I'll remove it, I can't get it to reach here without passing through the first `TSNonNullExpression` check and vice versa (if I disable the first `TSNonNullExpression` check)